R/checks.R

```diff
@@ -282,6 +282,58 @@ check_workflow <- function(x, pset = NULL, check_dials = FALSE) {
     }
   }
 
+  if (has_preprocessor_recipe(x)) {
+    recipe <- extract_recipe(x, estimated = FALSE)
+    exclusions <- c("case_weights", "outcome", "predictor")
+
+    original_var_data <- recipe$var_info %>%
+      dplyr::filter(source == "original", is.na(role) | !(role %in% exclusions))
+
+    non_standard_roles <- original_var_data$variable
+
+    vars_print <- glue::glue_collapse(
+      non_standard_roles,
+      sep = ", ", last = ", and "
+    )
+
+    msg <- glue::glue(
+      "The following columns in the recipe have non-standard roles: {vars_print}."
+    )
+
+    if (is.null(x$pre$actions$recipe$blueprint)) {
```

Comment:
@DavisVaughan is there a more proper way to extract the possible recipes blueprint of a workflow?